addons/thtracker/thtracker.lua

```diff
@@ -24,162 +24,83 @@
-windower.register_event('logout', dispose)
-windower.register_event('unload', thunload)
-
-
-windower.register_event('addon command', function(...)
-	local params = {...};
-	if #params < 1 then
-		return
+windower.register_event('incoming text', function(original, new, color)
+	original = original:strip_format()
+	if string.find(original, "Treasure Hunter") and not string.find(original, "TH Tracker:") then
+		name, count = original:match('Additional effect: Treasure Hunter effectiveness against the (.*) increases to (%d+)')
+		if not name and not count then
+			name, count = original:match('Additional effect: Treasure Hunter effectiveness against (.*) increases to (%d+)')
+		end		
 	end
-	if params[1] then
-		if params[1]:lower() == "help" then
-			print('th help : Shows help message')
-			print('th pos <x> <y> : Positions the list')
-			print('th hide : Hides the box')
-			print('th show : Shows the box')
-		elseif params[1]:lower() == "pos" then
-			if params[3] then
-				thbox.posx(tonumber(params[2]))
-				thbox.posy(tonumber(params[3]))
-				thbox:update()
-				--windower.text.set_location(box_name, posx, posy)
-			else
-				print('THTracker is currently at '..posx..', '..posy)
-			end
-		elseif params[1]:lower() == "hide" then
-			see = false
-			thbox:hide()
-		elseif params[1]:lower() == "show" then
-			see = true
-			thbox:show()
-		elseif params[1]:lower() == "current" then
-			print('Box text should be: '..thoutput)
-		elseif params[1]:lower() == "delete" then
-			--windower.text.delete(box_name)
-		elseif params[1]:lower() == "create" then
-			--windower.text.create(box_name)
-			initialize()
-		end
-	end	
-end)
-
-windower.register_event('incoming text', function(original, modified, mode)
-	if(see == true) then
-		original = original:strip_format()
-		count = 0
-		if string.find(original, "Treasure Hunter") and not string.find(original, "TH Tracker:") then
-			a,b,name,count = string.find(original,'Additional effect: Treasure Hunter effectiveness against the (.*)%s? increases to (%d+).')
-			if name == nil then
-				--log('No name')
-				a,b,name,count = string.find(original,'Additional effect: Treasure Hunter effectiveness against (.*)%s? increases to (%d+).')
-			end
-		end
-		
-		if string.find(original, "defeats") and not string.find(original, "TH Tracker:") then
-			a,b,deadmob = string.find(original,'%w+ defeats the (.*)\46')
-			if deadmob == nill then
-				a,b,deadmob = string.find(original,'%w+ defeats (.*).?')
-			end
-		end
-
-		if name ~= nil and count ~= 0 then
-			mob = name
-			thbox.boxtext = (' '..name..'\n TH: '..count)
-			--thbox:show()
-			deadmob = nill
-		end
-		
-		if deadmob ~= nil then
-			thbox.boxtext = 'No current mob'
-			--thbox:hide()
-			mob = nil
-			deadmob = nil
-		end
+	
+    if name and count then
+        mob = name
+        th:text(' '..name..'\n TH: '..count);
+        th:show()
+		mob = nil
+		deadmob = nil
+		count = nil
+    end
+
+    deadmob = original:match('%w+ defeats the (.*).')
```

Comment:
Just another local variable change for you..

`local deadmob = original:match('%w+ defeats the (.*).') or original:match('%w+ defeats (.*).')`